anvil-velocity/src/main/kotlin/org/anvilpowered/anvil/velocity/metrics/VelocityMetricService.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Anvil - AnvilPowered
+ *   Copyright (C) 2020
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+package org.anvilpowered.anvil.velocity.metrics
+
+import com.google.inject.Inject
+import com.velocitypowered.api.proxy.ProxyServer
+import java.io.BufferedReader
+import java.io.BufferedWriter
+import java.io.File
+import java.io.FileReader
+import java.io.FileWriter
+import java.io.IOException
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.ArrayList
+import java.util.Optional
+import java.util.UUID
+import java.util.regex.Pattern
+import java.util.stream.Collectors
+import org.anvilpowered.anvil.api.Environment
+import org.anvilpowered.anvil.api.Platform
+import org.anvilpowered.anvil.api.server.LocationService
+import org.anvilpowered.anvil.common.metrics.MetricService
+import org.bstats.MetricsBase
+import org.bstats.json.JsonObjectBuilder
+import org.slf4j.Logger
+
+
+class VelocityMetricService : MetricService {
+
+    @Inject
+    private lateinit var logger: Logger
+
+    @Inject
+    private lateinit var platform: Platform
+
+    @Inject
+    private lateinit var proxyServer: ProxyServer
+
+    @Inject
+    private lateinit var locationService: LocationService
+
+    private lateinit var metricsBase: MetricsBase
+    private var serverUUID: String? = null
```

Comment:
uuid is UUID?

Reply:
I left this how it was as I didn't want to stray too far from what bastian provided.